layers/meta-balena-jetson/recipes-support/hostapp-update-hooks/files/99-resin-bootfiles-agx-orin-devkit

```diff
@@ -38,21 +48,67 @@ check_disable_xudc()
     fi;
 }
 
+# While the UEFI firmware accepts loading the uefi bootloader
+# from any partition, it doesn't do the same for
+# the update capsule, which needs to reside on a partition
+# that has the boot,esp flags set. Since we originally
+# integrated this device type using the default xml
+# which sets these flags for the 'esp' partition only and
+# we preferred to include the efi bootloader in our
+# own 'resin-boot' partition instead, we do not have the flags
+# set by the nvidia flashing tools and have to
+# set them now. This way, on the next update the
+# UEFI firmware will not load the UEFI bootloader
+# from the boot partition, and instead will read the
+# capsule and perform the update.
+check_and_update_esp_flags() {
```

Comment:
TODO: We tested this hook from L4T 35.2.1 to 35.3.1, but we should also flash the first image for this DT on 35.1 and try the update

Reply:
This BUP bundle does not work from 35.1 (first release) to 35.3.1

Reply:
Updated to use legacy mode if performing a HUP from 35.1 to latest